packages/block-editor/src/components/block-popover/index.js

```diff
@@ -156,28 +158,35 @@ function BlockPopover(
 	}
 
 	return (
-		<Popover
-			ref={ mergedRefs }
-			animate={ false }
-			focusOnMount={ false }
-			anchor={ popoverAnchor }
-			// Render in the old slot if needed for backward compatibility,
-			// otherwise render in place (not in the default popover slot).
-			__unstableSlotName={ __unstablePopoverSlot || null }
-			placement="top-start"
-			resize={ false }
-			flip={ false }
-			shift={ shift }
-			{ ...props }
-			className={ classnames(
-				'block-editor-block-popover',
-				props.className
-			) }
-			variant="unstyled"
-		>
-			{ __unstableCoverTarget && <div style={ style }>{ children }</div> }
-			{ ! __unstableCoverTarget && children }
-		</Popover>
+		<>
+			<Popover.Slot name={ `inline-block-popover-${ id }` } />
```

Comment:
I hesitated about the "inline" prop but I added it, it seems a bit simpler than having to add slots everywhere.

Reply:
There is some ambiguity when there is a `Popover.__unstableSlotNameProvider` together with `inline` prop. Example:
```js
function MyPopover( { __unstablePopoverSlot } ) {
  return <Popover
    __unstableSlotName={ __unstablePopoverSlot }
    inline={ ! __unstablePopoverSlot }
  />;
}

function App() {
  return (
    <Popover.__unstableSlotNameProvider value="block-tools-after">
      <MyPopover />
    </Popover.__unstableSlotNameProvider>
  );
}
```
Here the `App`'s intent was to render the `MyPopover` in the `block-tools-after` slot, but something else will happen: it will be rendered inline, because the `inline` prop is `true`.

It comes down to whether `inline={ true }` means
- prefer inline: if there is no slot either passed as a prop or in context, render the popover inline rather than in the fallback portal.
- force inline: render the popover always inline, even though a slot is specified. Ignore the slot.

The current implementation is closer to "prefer inline", but not exactly: it doesn't take into account the slot in the context.

The context provider is used by the `RichText` component to render `FormatEdit` popovers.

Reply:
I'd go with "force inline" personally to avoid ambiguity.

Reply:
For me the code is forcing inline regardless of the context or the slot name prop and this is what I wanted to implement. So I'm not sure I understand.

Reply:
I'm sorry, I misread the `shouldRenderWithinSlot = slot.ref && ! inline` condition.

When forcing inline, however, the `inline={ ! __unstablePopoverSlot }` prop in `BlockPopover` and `BlockPopoverInbetween` is wrong, as it's forcing inline when a slot is provided by context.

In practice this probably never happens, if the `Popover.__unstableSlotNameProvider` is really used only by `RichText`. `RichText` attaches its popovers to the edited content, it doesn't know what a block is. So it will never be used together with `BlockPopover`.